Statistics.md

```diff
@@ -0,0 +1,85 @@
+<!--
+  - Licensed to the Apache Software Foundation (ASF) under one
+  - or more contributor license agreements.  See the NOTICE file
+  - distributed with this work for additional information
+  - regarding copyright ownership.  The ASF licenses this file
+  - to you under the Apache License, Version 2.0 (the
+  - "License"); you may not use this file except in compliance
+  - with the License.  You may obtain a copy of the License at
+  -
+  -   http://www.apache.org/licenses/LICENSE-2.0
+  -
+  - Unless required by applicable law or agreed to in writing,
+  - software distributed under the License is distributed on an
+  - "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  - KIND, either express or implied.  See the License for the
+  - specific language governing permissions and limitations
+  - under the License.
+  -->
+
+Parquet statistics definitions
+===
+This file contains the specification of all supported encodings.
+
+### Bloom filter statistics
+
+Supported Types: Integer
+
+The bloom filter is stored per block which is used to filter some blocks when reading the parquet file.
+
+#### Background for bloom filter
+In short, bloom filter maintains a bit set which is set to 1 for all bits. Once one entry added to the filter, it will use the hash functions to set several bits to 1. Using thge same hash functions,
+we can see if a new entry is already included checking all of the related bits value. If all of them are set to 1, it means this entry is already added, otherwise not.
+In this way we can use the bloom filter to skip some data in block level for parquet. From the Wikipedia, you can see the definition of bloom filter as follows:
+```An empty Bloom filter is a bit array of m bits, all set to 0.
+There must also be k different hash functions defined, each of which maps or hashes some set element to one of the m array positions with a uniform random distribution.```
+To construct a bloom filter, two parameters are required:
+* _m_: Number of bits
+* _k_: Number of hash function
+These two parameters could be approximated by two other parameters (The formulas are available in the [Wikipedia](https://en.wikipedia.org/wiki/Bloom_filter)):
+
+* _f_: abbreviation for false positive probability
+* _n_: the number of elements being filtered
+
+The code snippet for the data structure of bloom filter is mainly referred from [Apache Hive](https://hive.apache.org/) project and the implementation of bloom filter from [Guava](http://code.google.com/p/guava-libraries/).
+
+### The structure of bloom filter in parquet-format
+In the statistics, we have the following information for bloom filter.
+```
+list<i64> bitSet: the bit set storing the generated bit values using added entries
+i32 numBits; number of bit which is evaluated in the way discussed in the previous section
+i32 numHashFunctions; number of hash functions to generate the bit from the encountering entries
+```
+
+#### Implementation of the bloom filter in parquet-mr
```

Comment:
the spec bellow applies to all implementations. 
We should not have a specific mention of parquet-mr here.
Each implementation should refer to what is supported.